docs/concepts/tools/kubectl/object-management-using-declarative-config.md

```diff
@@ -0,0 +1,517 @@
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+...
+spec:
+  replicas: 3
+  revisionHistoryLimit: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.10
+      - name: nginx-helper-b
+        image: helper:1.3
+        args: ["run"]
+      - name: nginx-helper-c # This field was added
+        image: helper:1.3
+      - name: nginx-helper-d
+        image: helper:1.3
+```
+
+
+## **Warning:** considerations around defaulting performed in the server
+
+Defaulted fields are *not* re-defaulted unless they are explicitly cleared
+as part of the patch request.
+This may cause unexpected behavior for a fields defaulted to based
+on the values of other fields.  When the other fields are later changed
+the values defaulted from them will not be updated unless they are
+explicitly cleared.
+
+For this reason, it is recommended that certain fields defaulted
+by the server are explicitly defined in the object configuration even
+if the desired values match the server defaults.  This will make it
+easier to recognize conflicting values that will not be re-defaulted
+by the server.
+
+Example:
+
+1. User creates a Deployment without defining `strategy.type`
+2. Server defaults `strategy.type` to `RollingUpdate` and defaults `strategy.rollingUpdate` values
+3. User changes `strategy.type` to `Recreate`.  The `strategy.rollingUpdate` values remain at their defaulted values, though the server expects them to be cleared.
+  - If the strategy.rollingUpdate was defined in the config initially, it would have been more clear that it needed to be deleted.
+4. Apply fails because `strategy.rollingUpdate` is not cleared
+
+Recommendations for fields to explicitly define in the object configuration:
+
+- Selectors and PodTemplate Labels on any controller
+- Deployment RolloutStrategy
+
+### How to clear server defaulted fields or fields set by other writers
+
+As of Kubernetes 1.5, fields that do not appear in the configuration cannot be
+cleared.  For lists that are merged, this means elements of the list
+that do not appear in the configuration cannot be deleted.  The
+following workaround exist:
+
+Option 1: remove the field by directly editing the live object
+
+Option 2: remove the field through the configuration
+- Add the field to the configuration to match the live object
+- Apply the configuration - updates the annotation to include the field
+- Delete the field from the configuration
+- Apply the configuration - delete the field from the live object and annotation.
+
+{% comment %}
+TODO(1.6): Update this with the following for 1.6
+
+Fields that do not appear in the configuration can be cleared by
+setting their values to `null` and then applying the configuration.
+For fields defaulted by the server, this will trigger re-defaulting
+the values.
+{% endcomment %}
+
+## How to change ownership of a field between configuration and live writers
```

Comment:
We should clarify what is live writer. I think it refer to controllers like HPA.